en/tools/calc-size.md

```diff
@@ -0,0 +1,331 @@
+in vbytes is the same as the size in bytes.  When used as witness data
+in a segwit input, the size in vbytes is the size in bytes divided by
+four.
+
+- `OP_*` (1)---all current opcodes in Bitcoin are a single byte
+
+- `ecdsa_public_key` ({{page.size.ecdsa_pubkey}})---old wallets may use 65-byte public keys
+
+- `ecdsa_signature` ({{page.size.ecdsa_signature}}) (about half of all
+  signatures generated with a random nonce are this size, about half are
+  one byte smaller, and a small percentage are smaller than that)
+
+- `schnorr_public_key` ({{page.size.schnorr_public_key}})
+
+- `schnorr_signature` ({{page.size.schnorr_signature}})---may also be
+  one-byte longer for a non-default signature hash (sighash)
+
+- `public_key_hash` ({{page.size.public_key_hash}})
+
+- `script_hash` ({{page.size.script_hash_p2sh}})---**P2SH only**
+
+- `script_hash` ({{page.size.script_hash_p2wsh}})---**P2WSH only**
+
+<script>
+function calculateTotal() {
+  var types = new Array();
+
+  types["p2pkh"] = {
+    "input": (
+      {{page.size.outpoint}}
+      + {{page.size.scriptsig_length}}
+      + {{page.size.p2pkh_ss}}
+      + {{page.size.nsequence}}
+    ),
+    "output": (
+        {{page.size.nvalue}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2pkh_spk}}
+    ),
+    "segwit": false
+  };
+
+  types["p2wpkh"] = {
+    "input": (
+      {{page.size.outpoint}}
+      + {{page.size.scriptsig_length}}
+      + {{page.size.p2wpkh_witness}}
+      + {{page.size.nsequence}}
+    ),
+    "output": (
+        {{page.size.nvalue}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2wpkh_spk}}
+    ),
+    "segwit": true
+  };
+
+  types["p2sh2_3"] = {
+    "input": (
+        {{page.size.outpoint}}
+      + {{page.size.scriptsig_length}}
+      + {{page.size.p2sh23_ss}}
+      + {{page.size.nsequence}}
+    ),
+    "output": (
+        {{page.size.nvalue}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2sh23_spk}}
+    ),
+    "segwit": false
+  };
+
+  types["p2wsh2_3"] = {
+    "input": (
+        {{page.size.outpoint}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2wsh23_witness}}
+      + {{page.size.nsequence}}
+    ),
+    "output": (
+        {{page.size.nvalue}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2wsh23_spk}}
+    ),
+    "segwit": true
+  };
+
+  types["p2tr"] = {
+    "input": (
+        {{page.size.outpoint}}
+      + {{page.size.scriptsig_length}}
+      + {{page.size.p2tr_witness}}
+      + {{page.size.nsequence}}
+    ),
+    "output": (
+        {{page.size.nvalue}}
+      + {{page.size.scriptpubkey_length}}
+      + {{page.size.p2tr_spk}}
+    ),
+    "segwit": true
+  };
+
+  // Get the form
+  var form = document.forms["vbytescalc"];
+
+  // The input/output type
+  // TODO: maybe allow the input type to be different from the output
+  //       type, or multiple inputs of different types (or multiple
+  //       outputs of different types)
```

Comment:
That would be cool! I might do a follow up PR that adds this.

Reply:
That'd be great!  Also having someone work on this who actually knows JS would probably be very beneficial.  :-)